src/ci/docker/scripts/rfl-build.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -euo pipefail
+
+LINUX_VERSION=8f5b5f78113e
+
+# Build rustc and rustdoc
+../x.py --stage 1 build library rustdoc
+
+# Install rustup so that we can use the built toolchain easily
+curl --proto '=https' --tlsv1.2 -sSf -o rustup.sh https://sh.rustup.rs
+sh rustup.sh -y --default-toolchain stable --profile minimal
+
+source /cargo/env
+
+rustup toolchain link local ./build/x86_64-unknown-linux-gnu/stage1
+rustup default local
+
+mkdir -p rfl
+cd rfl
+
+# Download Linux
+git clone --depth 1 --branch ${LINUX_VERSION} \
+    https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+
+# Install bindgen. It needs to be compiled with the same version
+# as will be used for the RfL build.
```

Comment:
I had some issues with preinstalling bindgen. I need to clone Linux first to find the version of bindgen to install, but when I do download Linux in the Dockerfile, it's essentially read only (due to Docker root permissions), so I can't easily perform an in-place build inside of the downloaded directory. So unless there's an easy way to do a fully out-of-source build for RfL (without even creating the RfL config file in the Linux directory), I would just keep it like this for simplicity. The whole thing takes <15 minutes, even without a cached Docker build, so it should be fine.

Reply:
Out-of-tree kernel builds with read-only source trees are supported and easy to do, so we should be able to do everything in the `Dockerfile` (i.e. clone + `bindgen` with `--locked`), e.g.:

```sh
make -C $LINUX_FOLDER O=$BUILD_FOLDER LLVM=1 -j$(($(nproc) + 1)) \
    rustavailable \
    defconfig

$LINUX_FOLDER/scripts/kconfig/merge_config.sh -O $BUILD_FOLDER -m $BUILD_FOLDER/.config $SOME_PATH/rfl-for-rust-ci.config

make -C $LINUX_FOLDER O=$BUILD_FOLDER LLVM=1 -j$(($(nproc) + 1)) \
    olddefconfig \
    samples/rust/rust_minimal.o \
    samples/rust/rust_print.o \
    drivers/net/phy/ax88796b_rust.o
```

(Even if that were not the case, wouldn't copying the tree to a writable place be better than re-cloning from network and using `bindgen` without `--locked`? A third alternative could be fixing the version of `bindgen` so that we do not need the clone before it, allowing to move the `bindgen` to the `Dockerfile`).

Reply:
Yeah, if the Dockerfile is changed, or some other CI config files change, it will be rebuilt automatically on CI. Otherwise it is cached.